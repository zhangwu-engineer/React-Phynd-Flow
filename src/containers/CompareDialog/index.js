import React from 'react';
import Grid from '@material-ui/core/Grid';
import Dialog from '@material-ui/core/Dialog';
import DialogTitle from '@material-ui/core/DialogTitle';
import DialogContent from '@material-ui/core/DialogContent';
import DialogActions from '@material-ui/core/DialogActions';
import Typography from '@material-ui/core/Typography';
import StyledButton from 'components/StyledButton';
import ButtonGroup from '@material-ui/core/ButtonGroup';
import Tooltip from '@material-ui/core/Tooltip';
import { Column, Table } from 'react-virtualized';
import 'react-virtualized/styles.css';
// style
import useStyles from './style';
//heler
import { getNameFromID } from 'utils/helper';

const height = 400;
const rowHeight = 50;
const headerHeight = 30;
const width = 800;

const CompareDialog = ({
  isModalShown,
  stashesList,
  hideModal,
  submitCTA,
  submitOne,
  revertOne,
}) => {
  const classes = useStyles();
  const closeModal = () => {
    hideModal();
  }

  const submitStash = cellData => {
    submitOne(cellData);
  }

  const revertStash = cellData => {
    revertOne(cellData);
  }

  const cellButtonRenderer= ({ cellData, rowIndex, dataKey }) => (
    <ButtonGroup
      variant="contained"
      size="small"
      color="primary"
      aria-label="contained primary button group"
      className={classes.buttonGroup}
    >
      <StyledButton onClick={() => submitStash(cellData)}>Submit</StyledButton>
      <StyledButton onClick={() => revertStash(cellData)}>Revert</StyledButton>
    </ButtonGroup>
  )

  const cellTextRenderer= ({ cellData, rowIndex, dataKey }) => (
      <Tooltip title={cellData} placement="top-end">
        <Typography className={classes.stashColumn}>{getNameFromID(cellData)}</Typography>
      </Tooltip>
  )

  const cellNumberRenderer= ({ cellData, rowIndex, dataKey }) => (
    <Typography className={classes.stashColumn}>
      { parseFloat(cellData) > -1 ? cellData+1 : '' }
    </Typography>
)

  return (
    <Dialog
      open={isModalShown}
      onClose={closeModal}
      aria-labelledby="alert-dialog-title"
      aria-describedby="alert-dialog-description"
      maxWidth={null}
    >
      <DialogTitle>
        <Typography>Compare</Typography>
      </DialogTitle>
      <DialogContent className={classes.dialogContent} dividers>
        <Grid className={classes.tabInputContent} container>
          <Table
            width={width}
            height={height}
            headerHeight={headerHeight}
            rowHeight={rowHeight}
            rowCount={stashesList ? stashesList.length : 0}
            rowGetter={({index}) => stashesList[index]}
            className={classes.stashTable}
          >
            <Column
              label="Module"
              dataKey="module"
              width={150}
              cellRenderer={cellTextRenderer}
            />
            <Column
              className={classes.stashColumn}
              label="Entity"
              dataKey="entity"
              width={150}
              cellRenderer={cellTextRenderer}
            />
            <Column
              className={classes.stashColumn}
              label="Index"
              dataKey="panelIndex"
              width={60} 
              cellRenderer={cellNumberRenderer}
            />
            <Column
              className={classes.stashColumn}
              label="Name"
              dataKey="itemName"
              width={180}
              cellRenderer={cellTextRenderer}
            />
            <Column
              className={classes.stashColumn}
              label="Actions"
              dataKey="itemName"
              width={260}
              cellDataGetter={({ rowData }) => rowData}
              cellRenderer={cellButtonRenderer}
            />
          </Table>
        </Grid>
      </DialogContent>
      <DialogActions className={classes.dialogActions}>
        <StyledButton variant="contained" onClick={submitCTA} bgcolor="#577399">Submit All</StyledButton>
      </DialogActions>
    </Dialog>
  );
}

export default CompareDialog;