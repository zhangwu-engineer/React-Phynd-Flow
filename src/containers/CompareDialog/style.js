import { makeStyles } from '@material-ui/core/styles';

export default makeStyles(theme => ({
  dialogContent: {
    paddingTop: 20,
    paddingBottom: 12,
  },
  dialogActions: {
    paddingTop: 20,
    paddingBottom: 20,
  },
  tabInputContent: {
    paddingBottom: 40,
  },
  buttonGroup: {
    paddingTop: 20,
    paddingBottom: 20,
    paddingLeft: 0,
    paddingRight: 0,
    justifyContent: 'flex-end',
  },
  button: {
    textTransform: 'none',
    marginLeft: 20,
  },
  resize: {
    fontSize: 20
  },
  stashTable: {
    fontSize: 16,
    textAlign: "center",
  },
  stashColumn: {
    fontSize: 16,
  }
}));
