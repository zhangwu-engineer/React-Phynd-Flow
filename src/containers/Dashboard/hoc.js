import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';

import {
  getDashboardReducer,
  getFieldsList,
  getStashesList,
  makeSidebarData,
  makeDashboardList,
  makeBlockList,
  isContactMap,
} from '../../selectors';

import * as actions from 'actions';

export const hoc = (WrapperContainer) => {
  class ProvidersHoc extends Component {
    constructor(props) {
      super(props);
      props.getDashboardDataRequest();
      props.getFieldsPerEntityRequest();
    }

    updateDashboard = data => {
      this.props.updateDashboardDataRequest({ data });
    }

    updateFields = data => {
      this.props.updateFieldsDataRequest({ data });
    }

    stashData = data => {
      this.props.addStashesDataRequest({ data });
    }

    revertStore = () => {
      const data = this.props.dashboardReducer.origin;
      this.props.revertAllDashboardDataRequest({ data });
    }

    submitStore = () => {
      const data = this.props.dashboardReducer.dashboard;
      this.props.submitAllDashboardDataRequest({ data });
      
    }

    submitOne = (data) => {
      this.props.submitOneDashboardDataRequest({ data });
    }

    revertOne = (data) => {
      this.props.revertOneDashboardDataRequest({ data });
    }

    render() {
      return (
        <WrapperContainer
          history={this.props.history}
          dashboardReducer={this.props.dashboardReducer}
          dashboardList={this.props.dashboardList}
          fieldsList={this.props.fieldsList}
          isContactMap={this.props.isContactMap}
          updateDashboard={this.updateDashboard}
          updateFields={this.updateFields}
          stashData={this.stashData}
          revertCTA={this.revertStore}
          submitCTA={this.submitStore}
          submitOne={this.submitOne}
          revertOne={this.revertOne}
          classes={this.props.classes}
          width={this.props.width}
          {...this.props}
        />
      )
    }
  }

  ProvidersHoc.propTypes = {
    dashboardReducer: PropTypes.object,
    fieldsList: PropTypes.array,
    getProvidersRequest: PropTypes.func,
  }

  return withRouter(ProvidersHoc)
}

export default (WrapperContainer) => {
  const getSidebarData = makeSidebarData();
  const getDashboardList = makeDashboardList();
  const getBlockList = makeBlockList();

  const mapStateToProps = (state, props) => ({
    dashboardReducer: getDashboardReducer(state),
    dashboardList: getDashboardList(state, props),
    fieldsList: getFieldsList(state, props),
    stashesList: getStashesList(state),
    sidebarData: getSidebarData(state, props),
    blockList: getBlockList(state, props),
    isContactMap: isContactMap(state, props),
  })

  const mapDispatchToProps = {
    getDashboardDataRequest: actions.getDashboardDataRequest,
    updateDashboardDataRequest: actions.updateDashboardDataRequest,
    submitAllDashboardDataRequest: actions.submitAllDashboardDataRequest,
    submitOneDashboardDataRequest: actions.submitOneDashboardDataRequest,
    revertAllDashboardDataRequest: actions.revertAllDashboardDataRequest,
    revertOneDashboardDataRequest: actions.revertOneDashboardDataRequest,
    addStashesDataRequest: actions.addStashesDataRequest,
    updateFieldsDataRequest: actions.updateFieldsDataRequest,
    getFieldsPerEntityRequest: actions.getFieldsPerEntityRequest,
  }

  return connect(mapStateToProps, mapDispatchToProps)(
    hoc(WrapperContainer)
  )
}
