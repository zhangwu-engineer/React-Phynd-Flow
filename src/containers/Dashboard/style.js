import { makeStyles } from '@material-ui/core/styles';

export default makeStyles(theme => ({
  root: {
    display: 'flex',
    marginTop: 112,
  },
  content: {
    flexGrow: 1,
  },
  details: {
    width: 'calc(100vw - 279px)',
    overflow: 'scroll',
    flexDirection: 'column',
  },
  addButtonContainer: {
    margin: 24,
  },
  addButton: {
    fontSize: 22,
    marginRight: 20
  },
  toolbar: theme.mixins.toolbar,
}));
