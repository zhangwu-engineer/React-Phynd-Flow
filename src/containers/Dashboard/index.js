import React from 'react';
import { assign, map } from 'lodash';
import { DragDropContext, Droppable } from 'react-beautiful-dnd';
import Box from '@material-ui/core/Box';
import Typography from '@material-ui/core/Typography';
import Grid from '@material-ui/core/Grid';
import _ from 'lodash';

// Expansion Panel
import MuiExpansionPanel from '@material-ui/core/ExpansionPanel';
import MuiExpansionPanelSummary from '@material-ui/core/ExpansionPanelSummary';
import MuiExpansionPanelDetails from '@material-ui/core/ExpansionPanelDetails';

import Button from '@material-ui/core/Button';
import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
import CategoryDialog from 'containers/CategoryDialog';
import DetailsDialog from 'containers/DetailsDialog';
import CaseKeyDialog from 'containers/CaseKeyDialog';
import BlockListDialog from 'containers/BlockListDialog';
import OperationDialog from 'containers/OperationsDialog';
import PanelItem from 'components/PanelItem';
import Sidebar from 'components/Sidebar';

import {
  getNameFromID,
  getNameFromEntity,
  reorder,
} from 'utils/helper';
import hoc from './hoc';

// style
import useStyles from './style';


const refs = [];

const Panel = ({ items, startPoint, panelName, panelIndex, blockedItems, dashboardList, ...props }) => {
  const itemsList = items && map(items, (item, index) => {
    const panelInfo = panelIndex ? panelIndex.toString() : '';
    const indexInfo = index ? index.toString() : '';
    refs[panelInfo+indexInfo] = React.createRef();
    return {
      item,
      ref: refs[panelInfo+indexInfo],
      ...props
    }
  });
  return (
    <Box>
      {
        map(itemsList, (item, index) =>
          <PanelItem
            item={item}
            blockedItems={blockedItems}
            panelName={panelName}
            panelIndex={panelIndex}
            index={index}
            startPoint={startPoint}
            key={`${startPoint}-${panelIndex}-${index}`}
            dashboardList={dashboardList}
          />
        )
      }
    </Box>
  );
};

const Dashboard = ({ dashboardReducer, dashboardList, fieldsReducer, fieldsList, blockList, match, sidebarData, updateDashboard, updateFields }) => {
  const classes = useStyles();
  const [expanded, setExpanded] = React.useState(null);
  const [activePanel, setActivePanel] = React.useState(null);
  const [activeParent, setActiveParent] = React.useState(null);
  const [activeCard, setActiveCard] = React.useState(null);
  const [activeDetails, setActiveDetails] = React.useState(null);
  const [isCategoryModalShown, setCategoryModalShown] = React.useState(false);
  const [isDetailsModalShown, setDetailsModalShown] = React.useState(false);
  const [isCaseKeyModalShown, setCaseKeyModalShown] = React.useState(false);
  const [isBlockListModalShown, setBlockListModalShown] = React.useState(false);
  const [isOperationModalShown, setOperationModalShown] = React.useState(false);

  const handleChange = panel => (event, newExpanded) => {
    setExpanded(newExpanded ? panel : false);
  };

  const onDragEnd = result => {
    if (!result.destination || !result.source) {
      return;
    }

    const items = reorder(
      fieldsList,
      result.source.index,
      result.destination.index
    );

    const fieldsListFromReducer = assign({}, fieldsReducer);
    fieldsListFromReducer.fields[getNameFromID(match.params.module)][getNameFromID(match.params.entity)] = items;
    updateFields(fieldsListFromReducer.fields);
  }

  return (
    <div className={classes.root}>
      <Sidebar data={sidebarData} />
      <main className={classes.content}>
        <Typography
          component="div"
          role="tabpanel"
          id={`scrollable-auto-tabpanel-${0}`}
          aria-labelledby={`scrollable-auto-tab-${0}`}
        >
          {!Array.isArray(dashboardList) &&
            <DragDropContext onDragEnd={onDragEnd}>
              <Droppable droppableId="droppable">
                {provided => (
                  <div
                    {...provided.droppableProps}
                    ref={provided.innerRef}
                  >
                    <Panel
                      startPoint={0}
                      panelName={match.params.entity}
                      items={fieldsList}
                      classes={classes}
                      expanded={expanded}
                      blockedItems={blockList}
                      dashboardList={dashboardList}
                      setCategoryModalShown={setCategoryModalShown}
                      setDetailsModalShown={setDetailsModalShown}
                      setCaseKeyModalShown={setCaseKeyModalShown}
                      setOperationModalShown={setOperationModalShown}
                      setActivePanel={setActivePanel}
                      setActiveParent={setActiveParent}
                      setActiveCard={setActiveCard}
                      setActiveDetails={setActiveDetails}
                      updateDashboard={(payload) => {
                        const dashboardSource = assign({}, dashboardReducer);
                        dashboardSource.dashboard[getNameFromID(match.params.module)] = payload;
                        updateDashboard(dashboardSource.dashboard);
                      }}
                      handleChange={handleChange}
                    />
                    {provided.placeholder}
                  </div>
                )}
              </Droppable>
              <Grid className={classes.addButtonContainer}>
                <Button
                  variant="contained"
                  color="primary"
                  className={classes.addButton}
                  onClick={() => {
                    setBlockListModalShown(true);
                  }}
                >
                  Add Field
                </Button>
              </Grid>
            </DragDropContext>
          }
          {Array.isArray(dashboardList) && map(dashboardList, (dashboardItem, index) =>
            <DragDropContext onDragEnd={onDragEnd} key={`expansion-${index}`}>
              <Droppable droppableId="droppable">
                {(provided, snapshot) => (
                  <MuiExpansionPanel
                    square
                    {...provided.droppableProps}
                    ref={provided.innerRef}
                  >
                    <MuiExpansionPanelSummary
                      aria-controls={`panel-array${index}d-content`}
                      id={`panel-array${index}`}
                      expandIcon={dashboardItem && <ExpandMoreIcon />}
                    >
                      <Typography>{index+1}</Typography>
                    </MuiExpansionPanelSummary>
                    <MuiExpansionPanelDetails>
                      <Panel
                        startPoint={parseInt(dashboardItem.startPoint)}
                        panelName={match.params.entity}
                        panelIndex={index+1}
                        items={fieldsList}
                        classes={classes}
                        expanded={expanded}
                        blockedItems={blockList}
                        dashboardList={dashboardItem}
                        setCategoryModalShown={setCategoryModalShown}
                        setDetailsModalShown={setDetailsModalShown}
                        setCaseKeyModalShown={setCaseKeyModalShown}
                        setOperationModalShown={setOperationModalShown}
                        setActivePanel={setActivePanel}
                        setActiveParent={setActiveParent}
                        setActiveCard={setActiveCard}
                        setActiveDetails={setActiveDetails}
                        updateDashboard={(payload) => {
                          const dashboardSource = assign({}, dashboardReducer);
                          if (match.params.entity !== 'contacts') {
                            dashboardSource.dashboard[getNameFromID(match.params.module)][getNameFromEntity(match.params.entity)][index] = payload;
                          } else if (dashboardItem.addressIndex) {
                            dashboardSource.dashboard[getNameFromID(match.params.module)]['AddressMaps'][dashboardItem.addressIndex]['ContactMaps'][index] = payload;
                          }
                          updateDashboard(dashboardSource.dashboard);
                        }}
                        handleChange={handleChange}
                      />
                      {provided.placeholder}
                    </MuiExpansionPanelDetails>
                  </MuiExpansionPanel>
                )}
              </Droppable>
            </DragDropContext>
          )}
          {Array.isArray(dashboardList) &&
            blockList.length > 0 &&
            <Grid className={classes.addButtonContainer}>
              <Button
                variant="contained"
                color="primary"
                className={classes.addButton}
                onClick={() => {
                  setBlockListModalShown(true);
                }}
              >
                Add Field
              </Button>
            </Grid>
          }
        </Typography>
        <CategoryDialog
          isModalShown={isCategoryModalShown}
          hideModal={() => setCategoryModalShown(false)}
          activeParent={activeParent}
          currentCard={activeCard}
          currentDetails={activeDetails}
          setNewElement={(element) => {
            if (refs[activePanel])
              refs[activePanel].current.validateNew(element, activeParent);
          }}
          removeElement={() => {
            if (refs[activePanel])
              refs[activePanel].current.remove();
          }}
        />
        <DetailsDialog
          isModalShown={isDetailsModalShown}
          hideModal={() => setDetailsModalShown(false)}
          activeParent={activeParent}
          currentCard={activeCard}
          currentDetails={activeDetails}
          updateElement={(element, inputValue) => {
            if (refs[activePanel])
              refs[activePanel].current.validate(element, activeParent, inputValue);
          }}
          removeElement={() => {
            if (refs[activePanel])
              refs[activePanel].current.remove();
          }}
        />
        <CaseKeyDialog
          isModalShown={isCaseKeyModalShown}
          hideModal={() => setCaseKeyModalShown(false)}
          setNewCase={(inputKeyValue) => {
            if (refs[activePanel])
              refs[activePanel].current.validateCaseKey(activeParent, inputKeyValue);
          }}
        />
        <BlockListDialog
          isModalShown={isBlockListModalShown}
          hideModal={() => setBlockListModalShown(false)}
          blockedItems={blockList}
          addField={(fields) => {
            _.remove(blockList, function (item) {
              return fields.indexOf(item) > -1;
            });
          }}
        />
        <OperationDialog
          isModalShown={isOperationModalShown}
          hideModal={() => setOperationModalShown(false)}
          setNewOperation={(name, field, value) => {
            if (refs[activePanel])
              refs[activePanel].current.validateOperation(activeParent, name, field, value);
          }}
        />
      </main>
    </div>
  );
}

export default hoc(Dashboard);
