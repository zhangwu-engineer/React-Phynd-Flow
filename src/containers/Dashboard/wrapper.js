import React, { Fragment } from 'react';
import ReactNofitication from 'react-notifications-component';
import 'react-notifications-component/dist/theme.css';

import LayoutContainer from 'components/Layout';
import Dashboard from 'containers/Dashboard';
import CompareDialog from 'containers/CompareDialog';
import hoc from './hoc';

const WrapperContainer = ({
  history,
  dashboardReducer,
  dashboardList,
  fieldsList,
  isContactMap,
  stashesList,
  updateDashboard,
  stashData,
  updateFields,
  revertCTA,
  submitCTA,
  submitOne,
  revertOne,
  ...props
}) => {

  const [isCompareModalShown, setIsCompareModalShown] = React.useState(false);

  const compareStore = () => {
    setIsCompareModalShown(true);
  }

  return (
    <Fragment>
      <ReactNofitication />
      <LayoutContainer
        history={history}
        revertCTA={revertCTA}
        submitCTA={submitCTA}
        compareCTA={compareStore}
      />
      <Dashboard
        dashboardReducer={dashboardReducer}
        dashboardList={dashboardList}
        fieldsList={fieldsList}
        isContactMap={isContactMap}
        updateDashboard={updateDashboard}
        stashData={stashData}
        updateFields={updateFields}
        { ...props }
      />
      <CompareDialog
        isModalShown={isCompareModalShown}
        stashesList={stashesList}
        hideModal={() => setIsCompareModalShown(false)}
        submitCTA={submitCTA}
        submitOne={submitOne}
        revertOne={revertOne}
      />
    </Fragment>
  )
}

export default hoc(WrapperContainer);