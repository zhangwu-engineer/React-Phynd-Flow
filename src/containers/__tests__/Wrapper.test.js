import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { WrapperContainer } from 'containers/Dashboard/wrapper';
import LayoutContainer from 'components/Layout';
import Dashboard from 'containers/Dashboard';
import CompareDialog from 'containers/CompareDialog';
import ReactNofitication from 'react-notifications-component';

describe('To test the Wrapper Component functionality.', () => {
  configure({ adapter: new Adapter() });
  let wrapperComponent;
  const mockFn = jest.fn();
  let historyMock = { push: new mockFn() };
  const dashboardProps = {
    history: historyMock,
    dashboardReducer: {},
    dashboardList: [],
    fieldsList: [],
    isContactMap: true,
    updateDashboard: new mockFn(),
    updateFields: new mockFn(),
    stashData: new mockFn(),
    revertCTA: new mockFn(),
    submitCTA: new mockFn(),
    submitOne: new mockFn(),
    revertOne: new mockFn(),
  };

  beforeEach(() => {
    wrapperComponent = shallow(
      <WrapperContainer {...dashboardProps} />
    );
  });

  it("renders without crashing", () => {
    expect(wrapperComponent).not.toBe(null);
  });

  it('should genrate internal components correctly', () => {
    expect(wrapperComponent.find(ReactNofitication)).toHaveLength(1);
    expect(wrapperComponent.find(LayoutContainer)).toHaveLength(1);
    expect(wrapperComponent.find(CompareDialog)).toHaveLength(1);
    expect(wrapperComponent.find(Dashboard)).toHaveLength(1);
  });

  it('should transfer props to LayoutContainer correctly', () => {
    expect(wrapperComponent.find(LayoutContainer).props().history).toBe(dashboardProps.history);
    expect(wrapperComponent.find(LayoutContainer).props().revertCTA).toBe(dashboardProps.revertCTA);
    expect(wrapperComponent.find(LayoutContainer).props().submitCTA).toBe(dashboardProps.submitCTA);
  });

  it('should transfer props to Dashboard component correctly', () => {
    expect(wrapperComponent.find(Dashboard).props().dashboardReducer).toBe(dashboardProps.dashboardReducer);
    expect(wrapperComponent.find(Dashboard).props().dashboardList).toBe(dashboardProps.dashboardList);
    expect(wrapperComponent.find(Dashboard).props().fieldsList).toBe(dashboardProps.fieldsList);
    expect(wrapperComponent.find(Dashboard).props().isContactMap).toBe(dashboardProps.isContactMap);
    expect(wrapperComponent.find(Dashboard).props().updateDashboard).toBe(dashboardProps.updateDashboard);
    expect(wrapperComponent.find(Dashboard).props().stashData).toBe(dashboardProps.stashData);
    expect(wrapperComponent.find(Dashboard).props().updateFields).toBe(dashboardProps.updateFields);
  });

  it('should transfer props to CompareDialog correctly', () => {
    expect(wrapperComponent.find(CompareDialog).props().stashesList).toBe(dashboardProps.stashesList);
    expect(wrapperComponent.find(CompareDialog).props().submitCTA).toBe(dashboardProps.submitCTA);
    expect(wrapperComponent.find(CompareDialog).props().submitOne).toBe(dashboardProps.submitOne);
    expect(wrapperComponent.find(CompareDialog).props().revertOne).toBe(dashboardProps.revertOne);
  });

});
