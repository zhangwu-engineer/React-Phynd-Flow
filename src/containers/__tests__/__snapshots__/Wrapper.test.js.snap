// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`To test the Layout Component functionality. renders without crashing 1`] = `undefined`;

exports[`To test the Wrapper Component functionality. renders without crashing 1`] = `undefined`;
